cmd/panfs-helpers.go

```diff
@@ -14,3 +17,117 @@ func dotS3PrefixCheck(objects ...string) error {
 	}
 	return nil
 }
+
+// Publish source path to destination path, creates all the
+// missing parents if they don't exist and change the rights.
+func panfsPublishFile(srcFilePath, dstFilePath string, objMode, dirMode os.FileMode, ownerID, groupID int) (err error) {
+	if srcFilePath == "" || dstFilePath == "" {
+		return errInvalidArgument
+	}
+
+	if err = checkPathLength(srcFilePath); err != nil {
+		return err
+	}
+	if err = checkPathLength(dstFilePath); err != nil {
+		return err
+	}
+
+	if err = panOSMkdirAll(path.Dir(dstFilePath), dirMode, ownerID, groupID); err != nil {
```

Comment:
Do we really need to create all parent directories here? In most cases we explicitly create all needed directories before any operations. If there are no parent directory - it is ok to fail with an error.